### PR TITLE
CLDR-14647 Modern Dashboard Widget, updating...

### PR DESCRIPTION
**Pull request by btangmu**
-Make key and id unique for Dashboard rows; allow more than one row per xpath

-New module cldrDash.js for managing the Dashboard data

-Add xpathIndex to index all Dashboard rows by xpath and notification category

-Fix missing parameter tr for singleRowErrHandler in cldrTable.js

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-14647
- [x] Updated PR title and link in previous line to include Issue number

### CODE REVIEW

**Comment by btangmu** [comment]
This PR is to update counts/rows in Dashboard when the user votes. I think it accomplishes everything that the old Dashboard was able to do in that respect, plus a little more. If the user votes even on a row that wasn't in the dashboard, and the vote results in new errors/warnings/etc, the dashboard is updated accordingly; however the row may not appear in the right order in the dashboard since that ordering is currently only determined by the back end when it generates the complete dashboard json.

Further improvement of the updating might be facilitated by back-end changes. Note that we still depend on the deprecated SurveyAjax.getRow, which accepts a "dashboard=true" query parameter. The new API version doesn't accept that parameter and doesn't deliver the "json.issues" on which the dashboard-update code relies. Possibly "json.issues" could be reconstructed from the plain getrow data. The code would be cleaner, though, if "json.issues" had all the data needed for updating the dashboard for a single xpath -- just like a miniature version of the full dashboard json, in the same format but limited to a single path. The change here to VettingViewer.java doesn't accomplish anything yet other than to enabling to examine the full FileInfo data in the debugger; it points to where we generate something like a single-xpath set of dashboard data but then throw away everything except the labels like "Error".

File: tools/cldr-apps/js/src/esm/cldrDash.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    hex: xpstrid,
+  });
+  if (result && result.ph) {
+    // success
+    return result.ph;
+  }
+  // failure; fall back on dummy header, and current section/page
+  return {
+    header: "[header]",
+    page: cldrStatus.getCurrentPage(),
```
Comment (srl295) [inline tools/cldr-apps/js/src/esm/cldrDash.js:215]: Log err?

**Review by srl295 — APPROVED** [review]

File: tools/cldr-apps/js/src/esm/cldrDash.js (hunk 1 above)
Comment (btangmu) [reply tools/cldr-apps/js/src/esm/cldrDash.js:215]: Good idea. I'll add that. The fallback might be superfluous... This all needs more testing...
